plugins/repoclosure.py

```diff
@@ -86,9 +99,12 @@ def _get_unresolved(self, arch=None):
         if self.opts.check:
             checkpkgs = set()
             for repo in self.opts.check:
-                for pkgs_filtered in available.filter(reponame=repo):
+                for pkgs_filtered in self.base.sack.query().available().filter(reponame=repo):
```

Comment:
For my use case, I need two separate package lists:
1. This list of packages in my local repo, which are the items I wish to check dependencies of, even if there is a "later" package in one of the other repos, and
2: To check dependencies against, the latest version of a package over all of the available repos, as that is what dnf will try to pull in if I want to install a package as a dependency of one of my local packages.

A case I am trying to guard against is where I have a package A that requires package libB, both of which are in my local repo, but there could be a later version of libB in, say, Fedora, which might have a different soname and hence result in a broken dependency when attempting to do "dnf upgrade" after installing A and libB from my local repo.

So "checkpkgs" is the first list, and "available" is the second list in my approach.

I think there's merit in having a "--newest" option, but maybe it's intuitively the default case? Perhaps a "--allpkgs" option to check older packages as well as the newest ones, but I'm not sure what the use case for that would be?